final-project/static/scripts.js

```diff
@@ -33,25 +32,33 @@ $(function retrieveJSON() {
     var product_id = $(event.target).parent().attr("id");
     console.log("element id: " + product_id)
 
-    // Create parameters for Flask.url_for() method.
-    var parameters = {
-      id: product_id
-    }
-    console.log("parameters:")
-    console.log(parameters)
-
-    // Retrieve the desired product info.
-    getProductInfo(parameters, function(productInfo) {
-      // If no JSON was found, tell the user.
-      if (typeof(productInfo) == "undefined") {
-
-        // TODO: Return an error message to the window.
-        console.log("404 Error: Product support is on top of this (kind of)")
+    // If the user clicked on a product, return product data.
+      // Refactor: This conditional needs a longer term solution.
+    if ( Number(product_id) >= 0 && Number(product_id) <= 1000 ) {
```

Comment:
Why are these not coming back as numbers in the first place?

Reply:
IDs should actually be unique strings, and not chronological indexes.

Reply:
Why does the id even need to fall into this range?

Reply:
No reason, it just happened to work at the moment, because I only have id's ranging from 1-3.

I was avoiding pre-mature optimization 🥇